feature/container/containerz/tests/container_lifecycle/containerz_test.go

```diff
@@ -65,5 +110,156 @@ func TestDeployAndStartContainer(t *testing.T) {
 	if _, err = cntrCli.Ping(ctx, &cpb.PingRequest{}); err != nil {
 		t.Errorf("unable to reach cntrsrv: %v", err)
 	}
+}
+
+// TestRetrieveLogs implements CNTR-1.2 validating that logs can be retrieved from a
+// running container.
+func TestRetrieveLogs(t *testing.T) {
+	ctx := context.Background()
+	cli := startContainer(ctx, t)
+	defer stopContainer(ctx, t, cli)
+
+	logCh, err := cli.Logs(ctx, instanceName, false)
+	if err != nil {
+		t.Errorf("unable to obtain logs for %s: %v", instanceName, err)
+	}
+
+	var logs []string
+	for msg := range logCh {
+		logs = append(logs, msg.Msg)
+		if msg.Error != nil {
+			t.Errorf("logs returned an error: %v", err)
+		}
+	}
+
+	if len(logs) == 0 {
+		t.Errorf("no logs were returned")
+	}
+}
+
+// TestListContainers implements CNTR-1.3 validating listing running containers.
+func TestListContainers(t *testing.T) {
+	ctx := context.Background()
+	cli := startContainer(ctx, t)
+	defer stopContainer(ctx, t, cli)
+
+	listCh, err := cli.ListContainer(ctx, true, 0, nil)
+	if err != nil {
+		t.Errorf("unable to list containers: %v", err)
+	}
+
+	wantCntrs := []string{"cntrsrv:latest"}
+	var gotCntrs []string
+
+	for cnt := range listCh {
+		gotCntrs = append(gotCntrs, cnt.ImageName)
+	}
+
+	if diff := cmp.Diff(wantCntrs, gotCntrs, cmpopts.SortSlices(func(l, r string) bool { return l < r })); diff != "" {
+		t.Errorf("ListContainer() returned diff (-want, +got):\n%s", diff)
+	}
+}
+
+// TestStopContainer implements CNTR-1.4 validating that stopping a container works as expected.
+func TestStopContainer(t *testing.T) {
+	ctx := context.Background()
+	cli := startContainer(ctx, t)
+	stopContainer(ctx, t, cli)
+
+	// wait for container to stop
+	time.Sleep(2 * time.Second)
+
+	listCh, err := cli.ListContainer(ctx, true, 0, nil)
+	if err != nil {
+		t.Errorf("unable to list containers: %v", err)
+	}
+
+	for cntr := range listCh {
+		t.Errorf("StopContainer did not stop the container: %v", cntr)
+	}
+}
+
+// TestVolumes checks that volumes can be created or removed, it does not test
+// if they can actually be used.
+func TestVolumes(t *testing.T) {
+	ctx := context.Background()
+	cli := containerzClient(ctx, t)
+
+	wantVolume := "my-vol"
+	gotVolume, err := cli.CreateVolume(ctx, "my-vol", "local", nil, nil)
+	if err != nil {
+		t.Errorf("unable to create volume: %v", err)
+	}
+	defer cli.RemoveVolume(ctx, "my-vol", true)
+
+	if wantVolume != gotVolume {
+		t.Errorf("incorrect volume name: want %s, got %s", wantVolume, gotVolume)
+	}
+
+	t.Logf("created volume %s", gotVolume)
+
+	volCh, err := cli.ListVolume(ctx, nil)
+	if err != nil {
+		t.Errorf("unable to list volumes: %v", err)
+	}
+
+	wantVolumes := []*client.VolumeInfo{
+		{
+			Name:    "my-vol",
+			Driver:  "local",
+			Options: map[string]string{"device": "", "o": "", "type": "none"},
```

Comment:
Can we document what these `Options` here are please? Are these default labels that are added by Docker?